en/docs/develop/writing-a-custom-federated-authenticator.md

```diff
@@ -106,1262 +111,118 @@ public interface ApplicationAuthenticator extends Serializable {
-            
-                Twitter twitter = new TwitterFactory().getInstance();
-                twitter.setOAuthConsumer(apiKey, apiSecret);
-                
-                try {
-                    String callbackURL = resourceBundle.getString("Call_Back_URL");
-                    RequestToken requestToken = twitter.getOAuthRequestToken(callbackURL.toString());
-                    request.getSession().setAttribute("requestToken",requestToken);
-                    request.getSession().setAttribute("twitter",twitter);
-                    response.sendRedirect(requestToken.getAuthenticationURL());
-            
-                } catch (TwitterException e) {
-                    LOGGER.error("Exception while sending to the Twitter login page.", e);
-                    throw new AuthenticationFailedException(e.getMessage(), e);
-                } catch (IOException e) {
-                    LOGGER.error("Exception while sending to the Twitter login page.", e);
-                    throw new AuthenticationFailedException(e.getMessage(), e);
-                }
-                return;
-        }
-            
-        protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response, AuthenticationContext context) throws AuthenticationFailedException {
-                Twitter twitter = (Twitter) request.getSession().getAttribute("twitter");
-                RequestToken requestToken = (RequestToken) request.getSession().getAttribute("requestToken");
-                String verifier = request.getParameter("oauth_verifier");
-                try {
-                    AccessToken token=twitter.getOAuthAccessToken(requestToken, verifier);
-                    request.getSession().removeAttribute("requestToken");
-                    User user= twitter.verifyCredentials();
-                    buildClaims(user,context);
-                } catch (TwitterException e) {
-                    LOGGER.error("Exception while obtaining OAuth token form Twitter", e);
-                    throw new AuthenticationFailedException("Exception while obtaining OAuth token form Twitter",e);
-                }
-        }
-            
-        public void buildClaims(User user, AuthenticationContext context) {
-            
-                    context.setSubject(String.valueOf(user.getId()));
-                    Map<ClaimMapping, String> claims = new HashMap<ClaimMapping, String>();
-                    claims.put(ClaimMapping.build("name", "name", null,false), user.getName());
-                    claims.put(ClaimMapping.build("screen_name", "screen_name", null,false), user.getScreenName());
-                    claims.put(ClaimMapping.build("url", "url", null,false), user.getURL());
-            
-                    context.setSubjectAttributes(claims);
-        }
-        ```
+1. Access the Management console of the partner identity server.
+2. Then go to the Service Providers under the main tab. 
```

Comment:
2.Navigate to **Main** > **Identity** > **Service Providers** > **Add**. 
3. Add a Service Provider name and optionally, add a description. 
4. Click **Register**.

Reply:
Fixed https://github.com/wso2/docs-is/pull/2527/commits/ffc80dd6ac9088892baa7c56e5f03a9aa0768d3f